src/resources/Catalogs/CatalogContent.ts

```diff
@@ -24,4 +30,10 @@ export default class CatalogContent extends Resource {
             this.buildPath(`${CatalogContent.baseUrl}/${sourceId}/metadata`, objectType),
         );
     }
+
+    getMetadataV2(sourceId: string, params?: CatalogMetadataNameParams) {
+        return this.api.get<MetadataPageModel>(
+            this.buildPath(`${CatalogContent.baseUrl}/${sourceId}/metadata`, params),
```

Comment:
Since its the exact same endpoint as `getMetadata` but just different params, I find it odd to name this method with `V2`. Can we instead modify `getMetadata` to accept `ObjectType | CatalogMetadataNameParams` as second parameter?

Reply:
Inside Platform-client, do we usually write a single function for api call using the same endpoint? If that is the practice then I might have to modify the call endpoint inside `TailgateService` and `Tailgate`

Reply:
We don't have those rules clearly defined, its more a question of avoiding surprises as much as possible. I might only be me, but personally I find it surprising that one endpoint can return 2 completely different responses structures. That might be more a question for #api-standards thought.

I'll give you my stamp of approval, but I am still not sure what is the best way forward.